stixcore/products/level1/quicklook.py

```diff
@@ -0,0 +1,165 @@
+        control = vstack((self.control, other.control))
+        # control = unique(control, keys=['scet_coarse', 'scet_fine'])
+        # control = control.group_by(['scet_coarse', 'scet_fine'])
+
+        other.data['control_index'] = other.data['control_index'] + self.control['index'].max() + 1
+        data = vstack((self.data, other.data))
+        data = unique(data, keys='time')
+        # data = data.group_by('time')
+        unique_control_inds = np.unique(data['control_index'])
+        control = control[np.isin(control['index'], unique_control_inds)]
+
+        energies = unique(vstack(self.energies, other.energies))
+
+        return type(self)(service_type=self.service_type, service_subtype=self.service_subtype,
+                          ssid=self.ssid, control=control, data=data, energies=energies)
+
+    def __repr__(self):
+        return f'<{self.__class__.__name__}\n' \
+               f' {self.control.__repr__()}\n' \
+               f' {self.data.__repr__()}\n' \
+               f'>'
+
+    # @classmethod
+    # def from_fits(cls, fitspath):
+    #     header = fits.getheader(fitspath)
+    #     header_info = {'service_type': int(header.get('stype')),
+    #                    'service_subtype': int(header.get('sstype')),
+    #                    'pi1_val': int(header.get('ssid')),
+    #                    'spid': int(header.get('spid'))}
+    #     control = QTable.read(fitspath, hdu='CONTROL')
+    #     data = QTable.read(fitspath, hdu='DATA')
+    #     energies = QTable.read(fitspath, hdu='ENERGIES')
+    #     if header['level'] == 'L0':
+    #         obs_beg = Time(DateTime.from_string(header['DATE_OBS']).to_datetime())
+    #     elif header['level'] == 'L1':
+    #         obs_beg = Time(header['DATE_OBS'])
+    #     data['time'] = (obs_beg + data['time'])
+    #     return cls(**header_info, control=control, data=data, energies=energies)
+
+    def to_days(self):
+        days = set([(t.year, t.month, t.day) for t in self.data['time'].to_datetime()])
+        date_ranges = [(datetime(*day), datetime(*day) + timedelta(days=1)) for day in days]
+        for dstart, dend in date_ranges:
+            i = np.where((self.data['time'] >= dstart) &
+                         (self.data['time'] < dend))
+
+            data = self.data[i]
+            control_indices = np.unique(data['control_index'])
+            control = self.control[np.isin(self.control['index'], control_indices)]
+            control_index_min = control_indices.min()
+
+            data['control_index'] = data['control_index'] - control_index_min
+            control['index'] = control['index'] - control_index_min
+            yield type(self)(service_type=self.service_type, service_subtype=self.service_subtype,
+                             ssid=self.ssid, control=control, data=data,
+                             energies=self.energies)
+
+    def get_energies(self):
```

Comment:
is this used and persisted somewhere?

Reply:
It 'should' be written to the L1 fits file but I will check.

Reply:
It  is written to the fits files 👍